### PR TITLE
Fixed usage of reserved keywords as column names

### PR DESCRIPTION
**Pull request by blinkzz**
Wanted to thank all contributors for this useful project, keep it up 👍 
This PR fixes a few issues that i've encountered while working on internal tool used for table cloning
Both of them were discovered by parsing `create_table_query` of `system.tables`:
* Usage of reserved word for column name
* Type hint parsing of nested columns, because clickhouse stores `a.b[]` definition as `` a.b.:`Array(JSON)` ``

### CODE REVIEW

File: parser/parser_column.go [hunk 1]
```diff
@@ -556,9 +563,16 @@ func (p *Parser) isSelectItemTerminatorKeyword() bool {
...
 
 func (p *Parser) parseColumnExpr(pos Pos) (Expr, error) { //nolint:funlen
-	// Should parse the keyword as an identifier if the keyword is followed by
-	// `,`, `AS`, another clause-starter keyword, or end-of-statement (EOF or
-	// `;`). ClickHouse accepts most reserved words as bare column names in
+	// A keyword followed by a dot is unambiguously the left side of a
+	// qualified column reference, even when it is otherwise reserved (for
+	// example, `kill.item_id`).
+	if p.matchTokenKind(TokenKindKeyword) && p.peekTokenKind(TokenKindDot) {
+		return p.parseIdentOrFunction(pos)
```
[the remark this answers is not in the document]
Comment (blinkzz) [reply parser/parser_column.go]: Fixed, also updated existing test so it includes this edge case

**Review by git-hulk — APPROVED** [review]

**Comment by git-hulk** [comment]
@blinkzz Thanks for your fix.
